src/tiled/automapper.cpp

```diff
@@ -691,7 +691,7 @@ static bool compareLayerTo(const TileLayer *setLayer,
                 bool matchListNo  = false;
 
 
-                if (!setLayer->contains(x + offset.x(), y + offset.y())) {
+                if (!(setLayer->contains(x + offset.x(), y + offset.y()) || setLayer->map()->infinite())) {
```

Comment:
No no, it needs to have that check. I found it strange too, but it doesn't work if this check is removed.